src/Surfnet/StepupGateway/GatewayBundle/Service/StepUpAuthenticationService.php

```diff
@@ -180,18 +182,48 @@ public function resolveHighestRequiredLoa(
 
         $spConfiguredLoas = $serviceProvider->get('configuredLoas');
 
-        if (!$loaCandidates->contains($spConfiguredLoas['__default__'])) {
+        if (array_key_exists('__default__', $spConfiguredLoas) &&
+            !$loaCandidates->contains($spConfiguredLoas['__default__'])
+        ) {
             $loaCandidates->add($spConfiguredLoas['__default__']);
+            $this->logger->info(sprintf('Added SP\'s default Loa "%s" as candidate', $spConfiguredLoas['__default__']));
         }
 
-        $this->logger->info(sprintf('Added SP\'s default Loa "%s" as candidate', $spConfiguredLoas['__default__']));
+        if (count($spConfiguredLoas) > 1 && is_null($identityInstitution)) {
+            throw new RuntimeException(
+                'SP configured LOA\'s are applicable but the authenticating user has no ' .
+                'schacHomeOrganization in the assertion.'
+            );
+        }
+
+        // Load the institutions the user has vetted a second factor
+        $institutionsBasedOnVettedTokens = $this->determineInstitutionsByIdentityNameId($identityNameId);
+
+        if (!is_null($identityInstitution) && !empty($institutionsBasedOnVettedTokens)) {
+            $institutionMatches = $this->institutionMatchingHelper->findMatches(
```

Comment:
Perhaps move this to a private method, I found $institutionMatches in the same scope as $matchingInstitutions confusing.